python/ray/tests/test_asyncio.py

```diff
@@ -304,15 +304,22 @@ def interceptor(e):
     ray.worker._unhandled_error_handler = interceptor
     x1 = f.remote()
     del x1
-    wait_for_condition(lambda: num_exceptions == 1)
+    if client_test_enabled():
+        # In client mode, x1 may not have received the underlying exception
+        # when deleted, so no exception is raised.
+        prev_num_exceptions = num_exceptions
+    else:
+        wait_for_condition(lambda: num_exceptions == 1)
+        prev_num_exceptions = 1
 
     # Test we don't report handled exceptions.
     x1 = f.remote()
     with pytest.raises(ray.exceptions.RayError):
         await x1
     del x1
     await asyncio.sleep(1)
-    assert num_exceptions == 1, num_exceptions
+    assert num_exceptions == prev_num_exceptions, (num_exceptions,
```

Comment:
maybe explicitly write expected number of exceptions: num_exceptions received number of exceptions: prev_num_exceptions in the error message